### PR TITLE
Closes #58 Anonymous session support

### LINKED ISSUE #58: Support Anonymous Cart and Checkout
The SDK should be able to retrieve an anonymous `access_token`. It is a new endpoint on `auth.*`, but follows the semantics of the `client_credentials` flow. The `/me/carts/` and `/me/orders/` endpoints can be used afterwards.

`POST https://{host}/oauth/{projectKey}/anonymous/token?grant_type=client_credentials&scope={scope}` returns an `access_token` and a `refresh_token`. They should be handled similar to the tokens from the password flow.

Per default, the API generates an id for the anonymous token that is used across carts and orders. Optionally, the param `anonymous_id` can be specified to set a custom id. The API validates that this id has not been used yet.

The regular /me sign up [1] will automatically move the carts and orders from the anonymous token to the newly created customer. The SDK needs to do nothing new, just use the anonymous token.

Signing in with an anonymous token is a two-step process (like sign in):
* On `api.*` the customer is signed in
* On `auth.*` a password flow token is requested

For the first step, `POST /me/login` is to be used. Fields:
* `email` - String
* `password` - String
* `activeCartSignInMode` - [AnonymousCartSignInMode](http://dev.commercetools.com/http-api-projects-customers.html#anonymouscartsigninmode) - Optional

Otherwise `/me/login` behaves like [2] (i.e. fails on wrong email/password).

[1] http://dev.commercetools.com/http-api-projects-me-profile.html#create-customer-sign-up
[2] http://dev.commercetools.com/http-api-projects-customers.html#authenticate-customer-sign-in

### PR DESCRIPTION
**Pull request by nikola-mladenovic**
Documentation + more tests to come...

### CODE REVIEW

File: Source/AuthManager.swift [hunk 1]
```diff
@@ -212,6 +267,25 @@ public class AuthManager {
...
+        if let authUrl = anonymousSessionTokenUrl, authHeaders = authHeaders, scope = Config.currentConfig?.scope {
+            var parameters = ["grant_type": "client_credentials", "scope": scope]
+            if let anonymousId = anonymousId {
+                parameters["anonymous_id"] = anonymousId
+            }
+
+            Alamofire.request(.POST, authUrl, parameters: parameters, encoding: .URLEncodedInURL, headers: authHeaders)
+            .responseJSON(queue: dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), completionHandler: { response in
+                self.state = .AnonymousToken
+                self.handleAuthResponse(response, completionHandler: completionHandler)
```
Comment (cneijenhuis) [inline Source/AuthManager.swift]: Maybe, if this was successful, `self.anonymousId` should be unset? Because it won't work to use it for a second time.

Comment (nikola-mladenovic) [inline Source/AuthManager.swift]: That is true. On the other hand, I don't think the app which uses the SDK would intentionally want to request another anonymous session. WDYT? If we reset `self.anonymousId` on first success, and the app accidentally requests another pair of session tokens, there won't be any failure, and the chances are it won't be very easy to spot by developers. Let me know how do you feel about this?

Comment (cneijenhuis) [inline Source/AuthManager.swift]: If someone opens the app (first anonymous session), logs in, then logs out, the app would want a second anonymous session, I think?

File: Source/Commercetools.swift [hunk 2]
```diff
@@ -23,4 +23,40 @@ public var config: Config? {
...
+/**
+    This method should be used to override `anonymousSession` Bool parameter from the configuration and get new tokens.
+    Once this method is invoked, any previously logged in user will be logged out. In case there was an anonymous
+    session active, the refresh token will be removed, and the session will not be recoverable any more.
+    Most common use case for this method is user logout.
+
+    - parameter usingSession:       Bool parameter indicating whether anonymous session should be used.
+*/
+public func obtainAnonymousToken(usingSession usingSession: Bool, anonymousId: String? = nil) {
+    AuthManager.sharedInstance.obtainAnonymousToken(usingSession: usingSession, anonymousId: anonymousId)
```
Comment (cneijenhuis) [inline Source/Commercetools.swift]: 👍 for these 3!

File: Source/AuthManager.swift (hunk 1 above)
Comment (nikola-mladenovic) [inline Source/AuthManager.swift]: Good point 👍 

Will nil the `self.anonymousId` on successful token retrieval.

**Comment by nikola-mladenovic** [comment]
After using the `obtainAnonymousToken` method for a while, it seemed to me that in many use-cases the app will need to know how did the token retrieval go, so I added `completionHandler: (NSError?) -> Void`, so the caller can know if something went wrong and what was the problem (e.g bad internet or already taken ID).

**Comment by cneijenhuis** [comment]
LGTM 👍

**Comment by nikola-mladenovic** [comment]
> Endpoints from the Commercetools services can be consumed without Checkout (PlainToken), with Guest Checkout (AnonymousToken) or with a logged in customer (CustomerToken). According to the configuration, all interactions with the Commercetools platform will be performed with a PlainToken or a AnonymousToken per default.

It's very clear and understandable 👍

**Pull request merged**